test/src/runner/testRunnerScreenshotsOutput.js

```diff
@@ -0,0 +1,313 @@
+      response : JSON.stringify({
+        status: -1,
+        state: '',
+        code: '',
+        value: null,
+        errorStatus: '',
+        error: 'Unknown command 404 Not Found',
+        httpStatusCode: 404
+      })
+    });
+
+
+    return NightwatchClient.runTests(testsPath, settings)
+      .then(_ => {
+        return readDirPromise('screenshots/sample')
+          .catch(err => {
+            assert.ok(err);
+            assert.ok(err.message.includes('no such file or directory'));
+          });
+      });
+  });
+
+  it('does not take screenshot if screenshot is disabled', function () {
+
+    let testsPath = [
+      path.join(__dirname, '../../sampletests/withfailures')
+    ];
+
+    let settings = {
+      skip_testcases_on_fail: true,
+      selenium: {
+        port: 10195,
+        version2: true,
+        start_process: true
+      },
+      output_folder: 'output',
+      silent: true,
+      globals: {
+        waitForConditionPollInterval: 5,
+        waitForConditionTimeout: 5,
+        retryAssertionTimeout: 1,
+        reporter: function () {
+        }
+      },
+      output: false,
+      screenshots: {
+        enabled: false,
+        on_failure: true,
+        on_error: true,
+        path: 'screenshots'
+      }
+    };
+
+
+    return NightwatchClient.runTests(testsPath, settings)
+      .then(_ => {
+        return readDirPromise('screenshots/sample')
+          .catch((err) => {
+            assert.ok(err);
+            assert.ok(err.message.includes('no such file or directory'));
+          });
+      });
+  });
+
+  it('does not take screenshot if screenshot is enabled but on_failure is set to false', function () {
+
+    let testsPath = [
+      path.join(__dirname, '../../sampletests/withfailures')
+    ];
+
+    let settings = {
+      skip_testcases_on_fail: true,
+      selenium: {
+        port: 10195,
+        version2: true,
+        start_process: true
+      },
+      output_folder: 'output',
+      silent: true,
+      globals: {
+        waitForConditionPollInterval: 5,
+        waitForConditionTimeout: 5,
+        retryAssertionTimeout: 1,
+        reporter: function () {
+        }
+      },
+      output: false,
+      screenshots: {
+        enabled: true,
+        on_failure: false,
+        on_error: true,
+        path: 'screenshots'
+      }
+    };
+
+
+    return NightwatchClient.runTests(testsPath, settings)
+      .then(_ => {
+        return readDirPromise('screenshots/sample')
+          .catch((err) => {
+            assert.ok(err);
+            assert.ok(err.message.includes('no such file or directory'));
+          });
+      });
+  });
+
+
+});
+
+function readDirPromise(dirName) {
+  return new Promise(function(resolve, reject) {
+    fs.readdir(dirName, function(err, result) {
+      if (err) {
+        return reject(err);
+      }
+
+      resolve(result);
+    });
+  });
+}
+
+// util to replace deprecated fs.existsSync
+function fileExistsSync(path) {
```

Comment:
Use this one https://github.com/nightwatchjs/nightwatch/blob/main/lib/utils/index.js#L316

Reply:
But there's no reason to make it sync here, it can return a promise, just like the readDir above.